### PR TITLE
Transpose shared matrices to better cache locality

### PR DESCRIPTION
**Pull request by Jake-Moss**
By bringing the thread id to the 0-axis we are able to make the thread local slices continuous in memory. This should improve cache locality for better performance in tight loops.

Benchmarked on the Arkansas state wide network and saw ~10% improvement in overall skimming runtime.

### CODE REVIEW

**Comment by pedrocamargo** [comment]
Documentation is failing due to change in behavior in Shapely. I will update the documentation on a separate PR

**Review by jamiecook — APPROVED** [review]

**Comment by pedrocamargo** [comment]
This can be merged after merging #370 and re-running tests

**Pull request merged**
